sdk/bootstrap/ip.go

```diff
@@ -6,38 +6,42 @@ import (
 	"os"
 )
 
-// This helper function gives the current container's IP address, when run in a
-// given network, mnt and UTS namespace.  The function first checks the presence of
-// `LIBPROCESS_IP` if present it will return this as the container's IP address
-// else it will look at the `hostname` setup for the container to pick out the
-// IP address associated with the hostname.
-func ContainerIP() (containerIP net.IP, err error) {
-	ip := os.Getenv("LIBPROCESS_IP")
-	if ip != "" {
-		containerIP = net.ParseIP(ip)
-		if containerIP == nil {
-			err = fmt.Errorf("Invalid LIBPROCESS_IP found: %s", ip)
-			return
-		}
-
-		if containerIP.String() != "0.0.0.0" && containerIP.String() != "::" {
-			return
-		}
+var libprocessIP = "LIBPROCESS_IP"
+var mesosContainerIP = "MESOS_CONTAINER_IP"
+
+// ContainerIP This helper function gives the current container's IP address, when run in a
+// given network, mnt and UTS namespace.
+func ContainerIP() (net.IP, error) {
+	// First, try MESOS_CONTAINER_IP (DC/OS 1.10+, when running the default executor)
+	ip, err := validateIP(mesosContainerIP)
+	if ip != nil || err != nil {
+		return ip, err
+	}
+
+	// Fall through to LIBPROCESS_IP (< DC/OS 1.10, custom executor)
+	ip, err = validateIP(libprocessIP)
+	if ip != nil || err != nil {
+		return ip, err
+	}
+
+	return nil, fmt.Errorf("Neither MESOS_CONTAINER_IP nor LIBPROCESS_IP is set. Cannot determine IP.")
+}
+
+func validateIP(envvar string) (net.IP, error) {
+	ip := os.Getenv(envvar)
+	if ip == "" {
+		return nil, nil
 	}
 
-	// LIBPROCESS_IP is empty or is set to 0.0.0.0 or ::
-	hostName, err := os.Hostname()
-	if err != nil {
-		err = fmt.Errorf("Unable to retrieve hostname: %s", err)
-		return
+	validIP := net.ParseIP(ip)
+	if validIP == nil {
+		return nil, fmt.Errorf("Invalid %s found: %s", envvar, ip)
 	}
 
-	ipAddr, err := net.ResolveIPAddr("ip", hostName)
-	if err != nil {
-		err = fmt.Errorf("Unable to resolve hostname(%s): %s", hostName, err)
-		return
+	if validIP.String() == "0.0.0.0" || validIP.String() == "::" {
```

Comment:
This check here implies support for IPv6 addresses, but the tests cases below don't exercise this.

Reply:
That is true. I was lazy.